File: src/main/java/com/medicalsystem/service/impl/UserDetailsServiceImpl.java
package com.medicalsystem.service.impl;

import com.medicalsystem.model.ApplicationUser;
import com.medicalsystem.service.ApplicationUserService;
import lombok.AllArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
@AllArgsConstructor(onConstructor = @__(@Autowired))
public class UserDetailsServiceImpl implements UserDetailsService {

    private final ApplicationUserService userService;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        ApplicationUser user = userService.findByUsername(username);

        if (user == null) {
            throw new UsernameNotFoundException(username);
        }

        return new org.springframework.security.core.userdetails.User(user.getUsername(), user.getPassword(), new ArrayList<>());
    }
}

File: src/main/kotlin/com/medicalsystem/security/UserDetailsServiceImpl.kt
package com.medicalsystem.security

import com.medicalsystem.domain.ApplicationUser
import com.medicalsystem.service.ApplicationUserService
import org.springframework.security.core.userdetails.User
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl(private val userService: ApplicationUserService) : UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails {
        val user: ApplicationUser? = userService.findByUsername(username)
        user?.let { return User(user.username, user.password, user.getAuthorities()) }
        throw UsernameNotFoundException(username)
    }
}